packages/vuetify/src/composables/filter.ts

```diff
@@ -0,0 +1,63 @@
+
+// Utilities
+import { getPropertyFromItem, propsFactory, wrapInArray } from '@/util'
+import { computed } from 'vue'
+
+// Types
+import type { PropType, Ref } from 'vue'
+
+export function defaultFilter (text: string, query?: string) {
+  if (typeof query !== 'string') return true
+  if (typeof text !== 'string') return false
+
+  return text.toLocaleLowerCase().includes(query.toLocaleLowerCase())
+}
+
+export type FilterFunction = (item: any, query?: string) => boolean
+
+export interface FilterProps {
+  filterFn?: FilterFunction
+}
+
+// Composables
+export const makeFilterProps = propsFactory({
+  filterFn: {
+    type: Function as PropType<FilterFunction>,
+    default: defaultFilter,
+  },
+}, 'filter')
+
+export function useFilter (
+  props: FilterProps,
+  items: Ref<any[]>,
+  query?: Ref<string | number>,
+  filterKeys?: (string | string[]) | (FilterFunction | FilterFunction[]),
+) {
+  const keys = computed(() => wrapInArray(filterKeys))
```

Comment:
making this `computed` here has no effect. if you want `filterKeys` to be reactive then the function needs to take `Ref<(string | string[]) ...>`